### PR TITLE
Feature/add task #Issue1

### PR DESCRIPTION
**Pull request by grzegorzroczniak555**
Rafał, rate my code, please

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -19,19 +19,21 @@
     "@angular/platform-browser": "~7.2.0",
     "@angular/platform-browser-dynamic": "~7.2.0",
     "@angular/router": "~7.2.0",
+    "angular-in-memory-web-api": "^0.9.0",
```
Comment (rbobko93) [inline package.json]: Since we are not using in memory web api we can remove it.

File: src/app/app.component.html [hunk 2]
```diff
@@ -1,21 +1,2 @@
...
-    <h2><a target="_blank" rel="noopener" href="https://angular.io/cli">CLI Documentation</a></h2>
-  </li>
-  <li>
-    <h2><a target="_blank" rel="noopener" href="https://blog.angular.io/">Angular blog</a></h2>
-  </li>
-</ul>
-
-<router-outlet></router-outlet>
+<h1>{{title}}</h1>
+<app-task></app-task>
```
Comment (rbobko93) [inline src/app/app.component.html]: This one needs to be replaced with router-module.

File: src/app/app.module.ts [hunk 3]
```diff
@@ -3,14 +3,25 @@ import { NgModule } from '@angular/core';
...
+    AppComponent,
+    TaskComponent
   ],
   imports: [
     BrowserModule,
-    AppRoutingModule
+    AppRoutingModule,
+    ReactiveFormsModule,
+    HttpClientModule,
+    HttpClientInMemoryWebApiModule.forRoot(
```
Comment (rbobko93) [inline src/app/app.module.ts]: We can remove it since we are not using it.

File: src/app/task-in-memory-data.service.ts [hunk 4]
```diff
@@ -0,0 +1,18 @@
+import { Injectable } from '@angular/core';
+import { InMemoryDbService } from 'angular-in-memory-web-api';
+import { Task } from './task/task';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class TaskInMemoryDataService implements InMemoryDbService {
```
Comment (rbobko93) [inline src/app/task-in-memory-data.service.ts]: Remove this whole service.

File: src/app/task.service.ts [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+  providedIn: 'root'
+})
+export class TaskService {
+
+  tasks: Task[] = [];
+
+  constructor() { }
+
+
+  // getTasks(): Observable<Task[]> {
```
Comment (rbobko93) [inline src/app/task.service.ts]: Do not commit commented code.

File: src/app/task/mock-tasks.ts [hunk 6]
```diff
@@ -0,0 +1,5 @@
+import { Task } from './task';
+
+export const tasks: Task[] = [];
```
Comment (rbobko93) [inline src/app/task/mock-tasks.ts]: Remove this file, it is not used.

File: src/app/task/task.component.ts [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+
+  addTask() {
+    const name = this.taskForm.get('name').value;
+    const description = this.taskForm.get('description').value;
+    const task = new Task(1, name, description);
+    this.taskService.addTask(task)
+      .subscribe(tasks => {
+        this.tasks = tasks;
+      });
+    console.log('Adding works!');
```
Comment (rbobko93) [inline src/app/task/task.component.ts]: Remove console.logs if it works.

File: src/app/task/task.component.ts [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+  tasks: Task[] = [];
+
+  constructor(public taskService: TaskService) { }
+
+  ngOnInit() { }
+
+  addTask() {
+    const name = this.taskForm.get('name').value;
+    const description = this.taskForm.get('description').value;
+    const task = new Task(1, name, description);
```
Comment (rbobko93) [inline src/app/task/task.component.ts]: Id is always 1. We need them to be generated randomly or to be in order 1,2,3 etc.

File: src/app/task/task.ts [hunk 9]
```diff
@@ -0,0 +1,16 @@
+enum Status {
```
Comment (rbobko93) [inline src/app/task/task.ts]: Move status to its own file

File: src/app/app.component.html [hunk 10]
```diff
@@ -1,2 +1,3 @@
 <h1>{{title}}</h1>
 <app-task></app-task>
+<router-outlet></router-outlet>
```
Comment (rbobko93) [inline src/app/app.component.html]: It needs to replace app-task not be underneath it.

File: src/app/task/task.component.ts [hunk 11]
```diff
@@ -20,16 +20,20 @@ export class TaskComponent implements OnInit {
 
   ngOnInit() { }
 
+  getId(min, max) {
```
Comment (rbobko93) [inline src/app/task/task.component.ts]: Id should be generated in Task class constructor when new object is created.

Comment (grzegorzroczniak555) [reply src/app/task/task.component.ts]: Do you mean I have to call this function first time in component? or definition of it has to be there?

Comment (rbobko93) [reply src/app/task/task.component.ts]: It is in the component right now. It's just that front end shouldn't be responsible for generating id. I consider component file in as frontend in our case. Usually when you send post request and object has id it usually means that something is wrong. So, let's move in different place. Let's move genId method to the class where our list of tasks lives and invoke it there to set Task id before adding task to the list.

File: src/app/task.service.ts [hunk 12]
```diff
@@ -11,7 +11,14 @@ export class TaskService {
...
   constructor() { }
 
+  getId(min, max) {
+    min = Math.ceil(min);
+    max = Math.floor(max);
+    return Math.floor(Math.random() * (max - min + 1)) + min;
+  }
+
   addTask(task: Task): Observable<Task[]> {
+    task.id = this.getId(1, 500);
```
Comment (rbobko93) [inline src/app/task.service.ts]: You are not assigning id to the task. As a result all of your tasks dont have id now.

Comment (grzegorzroczniak555) [reply src/app/task.service.ts]: image
I don't understand, after using a console.lof for list, I see different number of id

Comment (rbobko93) [reply src/app/task.service.ts]: My bad. You are completely right.

**Pull request closed without merging**